mlir/include/mlir/Transforms/Inliner.h

```diff
@@ -27,6 +27,10 @@ class InlinerConfig {
 public:
   using DefaultPipelineTy = std::function<void(OpPassManager &)>;
   using OpPipelinesTy = llvm::StringMap<OpPassManager>;
+  using CloneCallbackTy =
+      std::function<void(OpBuilder &builder, Region *src, Block *inlineBlock,
+                         Block *postInsertBlock, IRMapping &mapper,
+                         bool shouldCloneInlinedRegion)>;
```

Comment:
```suggestion
  using CloneCallbackSigTy =
      void(*)(OpBuilder &builder, Region *src, Block *inlineBlock,
                         Block *postInsertBlock, IRMapping &mapper,
                         bool shouldCloneInlinedRegion);
  using CloneCallbackTy =
      std::function<CloneCallbackSigTy>;
```

Splitting it like this will allow to easily declare the inlineRegion function as taking a `function_ref< InlinerConfig::CloneCallbackSigTy>`.